packages/components/src/custom-select-control-v2/legacy-component/test/index.tsx

```diff
@@ -87,7 +87,73 @@ describe.each( [
 	[ 'Uncontrolled', UncontrolledCustomSelectControl ],
 	[ 'Controlled', ControlledCustomSelectControl ],
 ] )( 'CustomSelectControl (%s)', ( ...modeAndComponent ) => {
-	const [ , Component ] = modeAndComponent;
+	const [ mode, Component ] = modeAndComponent;
+
+	it( 'Should select the first option when no explicit initial value is passed without firing onChange', async () => {
+		const mockOnChange = jest.fn();
+		render( <Component { ...legacyProps } onChange={ mockOnChange } /> );
+
+		expect(
+			screen.getByRole( 'combobox', {
+				expanded: false,
+			} )
+		).toHaveTextContent( 'violets' );
```

Comment:
Just as my comment below - this might make sense to be:

```suggestion
		).toHaveTextContent( legacyProps.options[ 0 ].name );
```

We're also doing that in some of the other tests already, so it makes sense for consistency reasons.